config/application.rb

```diff
@@ -50,7 +50,8 @@ class Application < Rails::Application
     # CORS configuration; see also cors_preflight route
     config.middleware.insert_before 0, Rack::Cors, logger: (-> { Rails.logger }) do
       allow do
-        origins { |source, _env| Settings.web_origin.split(',').include?(source) }
+        regex = Regexp.new(Settings.web_origin_regex)
+        origins { |source, _env| Settings.web_origin.split(',').include?(source) || source.match?(regex) }
```

Comment:
This is now evaluating the fully qualified domain strings in `Settings.web_origin` and the regex domains in `Settings.web_origin_regex`
Any merit to converting all domains to the regex version and eliminating the `Settings.web_origin` strings? Or just leave both options?

Reply:
Yeah, I think it might be a little cleaner and simpler to have them all in the regex. The regex format that is being used is slightly different than what I am used to and I don't know how advanced you can get with a regex in a yaml file, but if you add them all in, the regex might look something like this (I think): 

web_origin_regex: \Ahttps?:\/\/(localhost:300\d|127\.0\.0\.1:300\d|www\.va\.gov)\z

regex = Regexp.new(Settings.web_origin_regex)
origins { |source, _env| source.nil? || source.match?(regex) }

Reply:
Will merge as-is for now.  Converting it over will have to be a multi-step task:
* merge this
* change configs to use regex
* then we can remove the `Settings.web_origin` if we want.

Reply:
drive-by comment: it might be "simpler" to have everything in one place, but I do worry about making that the default since there are more ways this can be implemented incorrectly with regexes than exact hostname strings. (and failures to do it correctly could be a potential security concern)